airgun/entities/domain.py

```diff
@@ -0,0 +1,192 @@
+        full_name = kwargs.get('full_name')
+        dns_capsule = kwargs.get('dns_capsule')
+        parameters = kwargs.get('parameters')
+        locations = kwargs.get('locations')
+        organizations = kwargs.get('organizations')
+
+        # Fill 'Domain' tab
+        if dns_domain:
+            view.domain.dns_domain.fill(dns_domain)
+        if full_name:
+            view.domain.full_name.fill(full_name)
+        if dns_capsule:
+            view.domain.dns_capsule.fill(dns_capsule)
+
+        # Fill 'Parameters' tab
+        if parameters:
+            params = []
+            for param_name, param_value in parameters.items():
+                params.append({'name': param_name, 'value': param_value})
+            view.parameters.params.fill(params)  # fill overwrites old params
+
+        # Fill locations tab
+        if locations:
+            view.locations.multiselect.set_assigned(locations)
+
+        # Fill organizations tab
+        if organizations:
+            view.organizations.multiselect.set_assigned(organizations)
+
+        view.submit_button.click()
+
+    def create(self, dns_domain, **kwargs):
+        """
+        Create a new domain.
+
+        :param dns_domain: str, full DNS domain name
+        :param full_name: (optional) str, full name describing the domain
+        :param dns_capsule: (optional) str of DNS capsule to use
+        :param parameters: (optional) param dict or list of param dicts to set
+            Format: [{'name': str, 'value': str}]
+        :param locations: list of location strings to be assigned.
+            All other locations will be unassigned.
+            Default: ['Default Location']
+        :param organizations: list of org strings to be assigned.
+            All other orgs will be unassigned.
+            Default: ['Default Organization']
+        """
+        view = self.navigate_to(self, 'New')
+
+        kwargs['dns_domain'] = dns_domain
+
+        if not kwargs.get('locations'):
+            kwargs['locations'] = [self._DEFAULT_LOC]
+        if not kwargs.get('organizations'):
+            kwargs['organizations'] = [self._DEFAULT_ORG]
+
+        self._submit_domain_form(view, **kwargs)
+
+        wait_for(
+            lambda: view.flash.messages, fail_condition=[], timeout=10,
+            delay=2, message='wait for flash success'
+        )
+        view.flash.assert_success_message(self._CREATE_SUCCESSFUL.format(dns_domain))
+
+    def search(self, value=None):
+        """Search for 'value' and return domain names that match.
+
+        :param value: text to filter (default: no filter)
+        """
+        view = self.navigate_to(self, 'All')
+        view.search(value or '')
+        return [row['Description'].text for row in view.table.rows()]
+
+    def read(self, domain_name):
+        """Return dict with properties of domain."""
+        view = self.navigate_to(self, 'Edit', entity_name=domain_name)
+        properties = dict(
```

Comment:
that data is also seem redundant as entity layer should know nothing about structure of the view, so we usually write:
```
        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
        return view.read()
```

Reply:
Ok, so I can move the collections of the properties here into a read() method under the Edit view if that works? Then here we'd just `return view.read()` as you suggest.